.github/workflows/cuda.yml

```diff
@@ -0,0 +1,71 @@
+name: CUDA GitHub Actions
+
+on:
+  pull_request_review_comment:
+    types: [created]
+
+jobs:
+  test:
+    name: CUDA
+    runs-on: [self-hosted, linux]
+    if: github.event.comment.body == '/gha run cuda-builds' && contains('OWNER,MEMBER,COLLABORATOR', github.event.comment.author_association)
+    timeout-minutes: 60
+    strategy:
+      fail-fast: false
+    steps:
+      - name: Setup or update software on host machine
+        run: |
+          sudo apt-get update
+          sudo apt-get install --no-install-recommends -y \
+              curl \
+              wget \
+              git \
+              apt-transport-https \
+              ca-certificates \
+              gnupg-agent \
+              software-properties-common
+          curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+          sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" -y
+          curl -sL https://nvidia.github.io/nvidia-docker/gpgkey | sudo apt-key add -
+          curl -sL https://nvidia.github.io/nvidia-docker/$(. /etc/os-release;echo $ID$VERSION_ID)/nvidia-docker.list | sudo tee /etc/apt/sources.list.d/nvidia-docker.list
+          sudo apt-get update
+          sudo apt-get install --no-install-recommends -y \
+              docker-ce \
+              docker-ce-cli \
+              containerd.io \
+              nvidia-docker2
+          sudo systemctl restart docker
+      - name: Remove old folder with repository
+        run: sudo rm -rf $GITHUB_WORKSPACE
```

Comment:
This step is needed because `actions/checkout@v1` fails to remove old files (CMake temporary build files particularly) from previous runs, because they were created in docker by another user.

```
Warning: Unable to run "git clean -ffdx" and "git reset --hard HEAD" successfully, delete source folder instead.
Error: One or more errors occurred. (One or more errors occurred. (Access to the path '/home/guoke/actions-runner/_work/LightGBM/LightGBM/build/CMakeFiles/3.18.1/CMakeCCompiler.cmake' is denied.)) (One or more errors occurred. (Access to the path '/home/guoke/actions-runner/_work/LightGBM/LightGBM/build/CMakeFiles/cmake.check_cache' is denied.)) (One or more errors occurred. (Access to the path '/home/guoke/actions-runner/_work/LightGBM/LightGBM/build/CMakeFiles/3.18.1/CMakeDetermineCompilerABI_C.bin' is denied.)) (One or more errors occurred. (Access to the path '/home/guoke/actions-runner/_work/LightGBM/LightGBM/build/CMakeFiles/CMakeError.log' is denied.)) (One or more errors occurred. (Access to the path '/home/guoke/actions-runner/_work/LightGBM/LightGBM/build/CMakeFiles/3.18.1/CMakeDetermineCompilerABI_CXX.bin' is denied.)) (One or more errors occurred. (Access to the path '/home/guoke/actions-runner/_work/LightGBM/LightGBM/build/CMakeCache.txt' is denied.)) (Access to the path '/home/guoke/actions-runner/_work/LightGBM/LightGBM/build/CMakeFiles/3.18.1/CMakeCCompiler.cmake' is denied.) (Access to the path '/home/guoke/actions-runner/_work/LightGBM/LightGBM/build/CMakeFiles/cmake.check_cache' is denied.) (Access to the path '/home/guoke/actions-runner/_work/LightGBM/LightGBM/build/CMakeFiles/3.18.1/CMakeDetermineCompilerABI_C.bin' is denied.) (Access to the path '/home/guoke/actions-runner/_work/LightGBM/LightGBM/build/CMakeFiles/CMakeError.log' is denied.) (Access to the path '/home/guoke/actions-runner/_work/LightGBM/LightGBM/build/CMakeFiles/3.18.1/CMakeDetermineCompilerABI_CXX.bin' is denied.) (Access to the path '/home/guoke/actions-runner/_work/LightGBM/LightGBM/build/CMakeCache.txt' is denied.)
Error: Exit code 1 returned from process: file name '/home/guoke/actions-runner/bin/Runner.PluginHost', arguments 'action "GitHub.Runner.Plugins.Repository.v1_0.CheckoutTask, Runner.Plugins"'.
```